fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Optimizer.java

```diff
@@ -509,6 +511,7 @@ private OptExpression logicalRuleRewrite(
             deriveLogicalProperty(tree);
         }
 
+        ruleRewriteDownTop(tree, rootTaskContext, OnPredicateMoveAroundRule.INSTANCE);
```

Comment:
Why needs to dow-top iterate? just avoid getting `deriveValueProperty ` multi times, any other reasons?

Reply:
select * from (select * from t1 where v1 = 1) join (select * from t2) on t1.v1 = t2.v1 join (select * from t3) on t2.v1 = t3.v1.  If we use the topdown way, we need use the rule iteratively. Use down top way, we just use the rule once. I think this maybe more efficient.